strider/query_planner.py

```diff
@@ -240,6 +285,50 @@ async def qg_to_og(qgraph):
     return ograph
 
 
+def fix_categories_predicates(query_graph):
+    """
+    Given a permuted query graph with one KP,
+    fix the node categories and predicates to match the KP
+
+    Returns false if there is a conflicting node type
+    """
+    for edge in query_graph['edges'].values():
+        kp = edge["kps"]
```

Comment:
Is edge["kps"] sometimes a list and sometimes a scalar? Looks like this comes from `permute_graph()`. It might be more readable if `permute_graph()` produced length-1 lists rather than scalars.